### PR TITLE
[releases/2.1] pre-publish steps

### PR DESCRIPTION
**Pull request by rajatkumar87**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Pre-publish steps for releases/2.1.

### Why should this Pull Request be merged?

Standard release procedure.

### What testing has been done?

N/A - PR

### CODE REVIEW

**Comment by dixonjoel** [comment]
I updated your PR description to give you an idea about what I'd expect for this type of PR.

File: packages/generator/pyproject.toml
Comment (dixonjoel) [inline packages/generator/pyproject.toml:1]: I think you need to remove the allow-prereleases here.

File: packages/generator/pyproject.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ grpcio = "^1.49.1"
 protobuf = "^4.21"
 black = ">=24.8.0"
 click-option-group = ">=0.5.6"
-ni-measurement-plugin-sdk-service = {version = "^2.1.0-dev1", allow-prereleases = true}
+ni-measurement-plugin-sdk-service = {version = "^2.1.0", allow-prereleases = true}
```
Comment (bkeryan) [inline packages/generator/pyproject.toml:29]: One of the things that is new for this release is that ni-measurement-plugin-sdk-generator now depends on ni-mesaurement-plugin-sdk-service.

This means we have to change the final release workflow:
- In the branch protection rules, we also need to remove check_nimg.yml from the list of required jobs.
- We need to remove the call to check_nimg.yml from https://github.com/ni/measurement-plugin-python/blob/main/.github/workflows/Publish_NIMS.yml

Cc: @dixonjoel

**Review by bkeryan — CHANGES_REQUESTED** [review]

File: packages/generator/pyproject.toml (hunk 1 above)
Comment (dixonjoel) [reply packages/generator/pyproject.toml:29]: I turned off the check_nimg requirement on the branch protection rules. I'll add some verbage to the release process indicating you need to remove and re-add both of those checks once the publish is done.

Comment (dixonjoel) [reply packages/generator/pyproject.toml:29]: @bkeryan I believe this PR will be ready to go in after #970. Resetting your vote and we won't check in before that PR.

Comment (bkeryan) [reply packages/generator/pyproject.toml:29]: and #971

Comment (bkeryan) [reply packages/generator/pyproject.toml:29]: @rajatkumar87 You skipped this step for ni-measurement-plugin-sdk-generator:

> Ensure that the file = "../.." dependencies in the dev dependency group have been commented out.

```
PS D:\dev\measurement-plugin-python> rg 'ni-measurement-plugin-sdk-service = \{path ='
packages\generator\pyproject.toml
40:ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\sample_measurement\pyproject.toml
19:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\niscope_acquire_waveform\pyproject.toml
21:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\game_of_life\pyproject.toml
19:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\nifgen_standard_function\pyproject.toml
21:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\nidcpower_source_dc_voltage\pyproject.toml
21:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\output_voltage_measurement\pyproject.toml
24:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\nidaqmx_analog_input\pyproject.toml
20:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\nidmm_measurement\pyproject.toml
21:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\nidcpower_source_dc_voltage_with_multiplexer\pyproject.toml
23:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\nivisa_dmm_measurement\pyproject.toml
22:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\niswitch_control_relays\pyproject.toml
21:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}

examples\nidigital_spi\pyproject.toml
21:# ni-measurement-plugin-sdk-service = {path = "../../packages/service", develop = true}
```

However, that kept the system tests passing and doesn't affect the distribution package so I'll ignore it.

FYI: If you did this for the examples, they would not work when users install them from the .zip file.

**Review by bkeryan — APPROVED** [review]

**Pull request merged**